staging/src/k8s.io/apimachinery/pkg/runtime/serializer/cbor/internal/modes/decode.go

```diff
@@ -22,6 +22,28 @@ import (
 	"github.com/fxamacker/cbor/v2"
 )
 
+var simpleValues *cbor.SimpleValueRegistry = func() *cbor.SimpleValueRegistry {
+	var opts []func(*cbor.SimpleValueRegistry) error
+	for sv := 0; sv <= 255; sv++ {
+		// Reject simple values 0-19, 23, and 32-255. The simple values 24-31 are reserved
```

Comment:
Looking at the cbor decoder, 23 undefined is treated similarly to 22 null: https://github.com/fxamacker/cbor/blob/02b69dbb52f4ecf450b3aa5e9a04b7a0b4bf409a/decode.go#L1556-L1558. Wouldn't we want to keep that behavior?

Reply:
The true, false, and null simple values have clear parallels with JSON, but JSON doesn't have a representation for undefined. Accepting undefined and treating it as null means that we'd forever support two ways to represent something that only has one JSON representation. Rejecting undefined forces encoders to write null if they mean null.